### ISSUE TITLE
RuboCop::Cop::Style::OrAssignment could also work for setting other variables

### ISSUE THREAD

**Issue by arjan0307** [rubocop-hq/rubocop#5803]
The Cop only works if you are changing the same variable i.e. 
```name = name ? name : 'Bozhidar'```,
but not when setting another
```foo = name ? name : 'Bozhidar'```.

## Expected behavior

When having this line of code:

```foo = name ? name : 'Bozhidar'```

I would expect to get an `RuboCop::Cop::Style::OrAssignment` error.

## Actual behavior

I get no error.

## Steps to reproduce the problem

Given the following line of code ```foo = name ? name : 'Bozhidar'``` no error is being raised when running rubocop

## RuboCop version

```
0.54.0 (using Parser 2.5.0.5, running on ruby 2.5.1 x86_64-darwin17)
```

**Comment by rrosenblum** [comment]
I like the idea behind this. Correct `foo = name ? name : 'Bozhidar'` to `foo = name || 'Bozhidar'`. The intention of the `Style/OrAssignment` cop is to identify places where `||=` can be used so this goes outside of its initial scope. I'm not sure whether we would want to add this functionality to `Style/OrAssignment` or create a different cop to handle it. I think the new functionality is inline with the name of the cop.
